File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int left = 0, count = 0;
        HashSet<Character> set = new HashSet<>();
        for(int right = 0; right < s.length(); right++){
            while(set.contains(s.charAt(right))){
                set.remove(s.charAt(left));
                left++;
            }
            set.add(s.charAt(right));
            count = Math.max(count, right-left+1);
        }
        return count;
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let l = 0;
    let max = 0;
    let set = new Set();

    for(let r=0; r<s.length; r++){
        while(set.has(s[r])){
            set.delete(s[l]);
            l++;
        }
        max = Math.max(max, (r-l)+1);
        set.add(s[r]);
    }
    return max;
};